pulsar_devkit/simple_examples/HLSIPs/src/simple_algo_mt.cpp

```diff
@@ -0,0 +1,12 @@
+#include "simple_algo_mt.h"
+#include <cmath>
+#include <cassert>
+#ifndef __SYNTHESIS__
+#include <cstdio>
+#endif
+
+void simple_algo_mt_hw( ap_int<8> inA1, ap_int<8> inA2, ap_int<8>inA3, ap_int<8>inA4, ap_int<8> inB1, ap_int<8> inB2, ap_int<8> inB3, ap_int<8> inB4, ap_int<32> &outA ) {
+
+  //		outA = inA + inB;
+  outA = sqrt(2*inA1*inB1*(coshf(inA2-inB2) - cos(inA3-inB3)));
```

Comment:
what's the resource usage on this guy?  would be fun to have a few examples of LUTs for cosh?

Reply:
![resources_invmass](https://user-images.githubusercontent.com/10013416/31081002-b83c12d8-a78a-11e7-8bfb-1d34e23005b9.png)

Reply:
Interesting, already something pretty simple uses up non-trivial amount of LUTs.  I guess a jet mass calculation could reuse the same LUT....